test/cuda/cufinufft1d_test.cu

```diff
@@ -50,6 +51,16 @@ int run_test(int method, int type, int N1, int M, T tol, T checktol, int iflag,
       fk[i].real(randm11());
       fk[i].imag(randm11());
     }
+  } else if (type == 3) {
+    for (int i = 0; i < M; i++) {
+      c[i].real(randm11());
+      c[i].imag(randm11());
+    }
+    s.resize(N1);
+    for (int i = 0; i < N1; i++) {
+      s[i] = M_PI * randm11();
```

Comment:
In test/finufft1d_test.cpp the "s" array is scaled by S = N1/2.  I suggest you do this here too, to make the space-freq product grow (hence the FFT size) as the number of modes. This matches CPU testers. Also for 2d, 3d, below.